### PR TITLE
feature(RC-21569): adds new sort icon and updated MenuItem styles

### PR DESCRIPTION
**Pull request by michael-alterio**
- Updated background-color and color for MenuItem default, selected and hover states

**MenuItem**

**Default Color**
image

**Custom Color**
image

**With Icon**
image

### CODE REVIEW

**Review by pcln-james — CHANGES_REQUESTED** [review]
removing the ability to customize colors for MenuItem would be a breaking change. Is there a way that we can have your new changes will keeping what's there today?

**Comment by wesleymartin85** [comment]
@michael-alterio before we merge this lets meet to update the styles

File: packages/menu/src/MenuItem/MenuItem.js
Comment (pcln-brianna) [inline packages/menu/src/MenuItem/MenuItem.js:27]: I think we still need to allow for custom colours to be set on the button's hover state to be backwards compatible. Also since this is changing from the default `primary.base` on hover it might still be worth making it a breaking/major change.

Screen Shot 2022-09-26 at 4 46 03 PM
Screen Shot 2022-09-26 at 4 46 22 PM

Comment (michael-alterio) [reply packages/menu/src/MenuItem/MenuItem.js:27]: Updated to major change! Let me discuss with Wes and other members of design-system team on how we want to handle the custom color hovering.

Comment (michael-alterio) [reply packages/menu/src/MenuItem/MenuItem.js:27]: @wesleymartin85 @sdalonzo @pcln-james How should we best support custom hover color while also supporting these new changes? (FYI: I made this a major change)

Comment (pcln-brianna) [reply packages/menu/src/MenuItem/MenuItem.js:27]: Discussed offline and we're good with only selected colour customization going forward (based on existing implementations getting the in-path update) + also setting it back down to a minor, thanks!

It might be good to split this up into two PRs (move icon changes into a new PR, keep this for the menu update) so that your icon update isn't blocked by testing this.

**Review by pcln-brianna — APPROVED** [review]

**Review by pcln-james — APPROVED** [review]

**Pull request merged**
